mirage/tar_mirage.ml

```diff
@@ -427,9 +455,119 @@ module Make_KV_RW (CLOCK : Mirage_clock.PCLOCK) (BLOCK : Mirage_block.S) = struc
   let rename _ ~source:_ ~dest:_ =
     Lwt.return (Error `Append_only)
 
-  let set_partial _ _ ~offset:_ _ =
-    Lwt.return (Error `Append_only)
+  let set_partial t key ~offset data =
+    let ( >>>= ) = Lwt_result.bind in
+    Lwt_mutex.with_lock t.write_lock (fun () ->
+        if Optint.Int63.(compare offset zero < 0) then
+          invalid_arg "Tar_mirage.set_partial: negative offset";
+        Lwt.return begin match get_node t.map key with
+          | Ok Value (hdr, offset) -> Ok (hdr, offset)
+          | Ok Dict _ -> Error `Path_segment_is_a_value
+          | Error _ as e -> e
+        end >>>= fun (hdr, data_offset) ->
+        (* FIXME: check more thoroughly offset *)
+        let offset = Optint.Int63.to_int64 offset in
```

Comment:
I'm not sure what I had in mind with this comment. I suspect I was concerned about signedness? Probably we can remove that comment.